presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliTests.java

```diff
@@ -141,6 +141,30 @@ public void shouldRunQueryFromFile()
         assertThat(trimLines(presto.readRemainingOutputLines())).containsAll(nationTableBatchLines);
     }
 
+    @Test(groups = CLI, timeOut = TIMEOUT)
+    public void shouldStopOnError()
+            throws IOException, InterruptedException
+    {
+        File temporayFile = File.createTempFile("test-sql", null);
+        temporayFile.deleteOnExit();
+        Files.write("select * from hive.default.nations;\nselect * from hive.default.nation;\n", temporayFile, UTF_8);
+
+        launchPrestoCliWithServerArgument("--file", temporayFile.getAbsolutePath());
```

Comment:
Current tests only test the stop on error behavior while a good chunk of the functionality we add here is about the exit code. So we should also check the exit code of the cli process.